recipes/tcl/8.6.9/conanfile.py

```diff
@@ -0,0 +1,220 @@
+        for root, _, files in os.walk(self.build_folder):
+            if "Makefile" in files:
+                tools.replace_in_file(os.path.join(root, "Makefile"), "-Dstrtod=fixstrtod", "", strict=False)
+
+        with tools.chdir(self.build_folder):
+            autoTools.make()
+
+    def build(self):
+        self._fix_sources()
+        if self.settings.compiler == "Visual Studio":
+            # do not treat warnings as errors 
+            tools.replace_in_file(os.path.join(self._source_subfolder, "win", "rules.vc"), "cwarn = $(cwarn) -WX", "")
+            self._build_nmake()
+        else:
+            self._build_autotools()
+
+    def package(self):
+        if self.settings.compiler == "Visual Studio":
+            self._build_nmake("install")
+        else:
+            with tools.chdir(self.build_folder):
+                autoTools = self._get_auto_tools()
+                autoTools.install()
+                autoTools.make(target="install-private-headers")
+            pkgconfig_dir = os.path.join(self._install_folder, "lib", "pkgconfig")
+            if os.path.isdir(pkgconfig_dir):
+                shutil.rmtree(pkgconfig_dir)
+        self.copy(pattern="license.terms", dst="licenses", src=self._source_subfolder)
+
+        tclConfigShPath = os.path.join(self._install_folder, "lib", "tclConfig.sh")
+        package_path = os.path.join(self._install_folder)
+        if self._is_mingw_windows:
+            package_path = package_path.replace("\\", "/")
+        tools.replace_in_file(tclConfigShPath,
+                              package_path,
+                              "${TCL_ROOT}")
+        tools.replace_in_file(tclConfigShPath,
+                              "\nTCL_BUILD_",
+                              "\n#TCL_BUILD_")
+
+        tools.replace_in_file(tclConfigShPath,
+                              "\nTCL_SRC_DIR",
+                              "\n#TCL_SRC_DIR")
+
+    def package_info(self):
+        libs = []
+        systemlibs = []
+        libdirs = []
+        for root, _, _ in os.walk(os.path.join(self._install_folder, "lib"), topdown=False):
+            newlibs = tools.collect_libs(self, root)
+            if newlibs:
+                libs.extend(newlibs)
+                libdirs.append(root)
+        if self.settings.os == "Windows":
+            systemlibs.extend(["ws2_32", "netapi32", "userenv"])
+        else:
+            systemlibs.extend(["m", "pthread", "dl"])
+
+        defines = []
+        if not self.options.shared:
+            defines.append("STATIC_BUILD")
+        self.cpp_info.defines = defines
+
+        self.cpp_info.bindirs = [os.path.join("bin", "bin")]
+        self.cpp_info.libdirs = libdirs
+        self.cpp_info.libs = libs
+        self.cpp_info.system_libs = systemlibs
+        self.cpp_info.includedirs = [os.path.join("bin", "include")]
+
+        if self.settings.os == "Macos":
+            self.cpp_info.exelinkflags.append("-framework Cocoa")
```

Comment:
https://docs.conan.io/en/latest/howtos/link_apple_framework.html#how-to-link-with-apple-frameworks